libraries/AP_NavEKF/EKFGSF_yaw.cpp

```diff
@@ -0,0 +1,678 @@
+/*
+  Definition of functions used to provide a backup estimate of yaw angle
+  that doesn't use a magnetometer. Uses a bank of 3-state EKF's organised
+  as a Guassian sum filter where states are velocity N,E (m/s) and yaw angle (rad)
+
+  Written by Paul Riseborough <p_riseborough@live.com.au>
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+
+#include "AP_NavEKF/EKFGSF_yaw.h"
+#include <AP_AHRS/AP_AHRS.h>
+#include <AP_Vehicle/AP_Vehicle.h>
+#include <GCS_MAVLink/GCS.h>
+
+EKFGSF_yaw::EKFGSF_yaw()
+{
+	// this flag must be false when we start
+	ahrs_tilt_aligned = false;
+
+	// these objects are initialised in initialise() before being used internally, but can be reported for logging before then 
+	memset(&GSF, 0, sizeof(GSF));
+	memset(&EKF, 0, sizeof(EKF));
+};
+
+void EKFGSF_yaw::update(const Vector3f delAng,
```

Comment:
the two Vector3f should be const references (avoids copy)

Reply:
Fixed